pages/signup/index.tsx

```diff
@@ -39,22 +39,46 @@ const ADD_TALENT = gql`
   }
 `;
 
+// Min 8 characters. At least 1 capital, 1 number
+const strongCombination = new RegExp(/^(?=.*?[0-9])(?=.*?[A-Z]).{8,}$/);
+
 const SignUpPage = ({ t }: PageProps): React.ReactElement => {
   const [formValues, setFormValues] = useState<SignupFormValues>(
     defaultSignupFormValues,
   );
 
   const router = useRouter();
+  const [weakPassword, setWeakPassword] = useState(false);
   const [passwordsIdentical, setPasswordsIdentical] = useState(true);
-  const [createUser] = useMutation(ADD_EMPLOYER);
-  const [createTalent] = useMutation(ADD_TALENT);
   const [passwordRepeat, setPasswordRepeat] = useState<Record<string, unknown>>(
     { passwordConfirm: '' },
   );
   const [error, setError] = useState<FirebaseError | null>(null);
 
+  const [createUser] = useMutation(ADD_EMPLOYER);
+  const [createTalent] = useMutation(ADD_TALENT);
   const auth = useAuth();
 
+  const checkPasswordStrength = (): boolean => {
+    const isPasswordWeak = !strongCombination.test(formValues.password);
+    setWeakPassword(isPasswordWeak);
+    return isPasswordWeak;
+  };
+
+  const handlePasswordChange = (
+    e: React.ChangeEvent<HTMLTextAreaElement>,
+  ): void => {
+    setWeakPassword(false);
+    setFormValues((oldValues) => {
+      const [propArray] = getPropArray('password');
+      const newValue = e.target.value;
+      return {
+        ...oldValues,
+        ...computeNestedValue(oldValues, propArray, newValue),
```

Comment:
I am getting a type error in line 77. Do you have an idea how to solve this one?

`Argument of type 'SignupFormValues' is not assignable to parameter of type 'Record<string, unknown>'.
  Type 'NewBaseUser & { type: UserType; }' is not assignable to type 'Record<string, unknown>'.
    Index signature is missing in type 'NewBaseUser & { type: UserType; }'.`

Reply:
The problem is with using Record<string, unknown> with an interface.: https://github.com/microsoft/TypeScript/issues/15300

It can be fixed by making SignupFormValues into a type. If you replace the end of the types file with this:
```

export type BaseUser = {
  gender: Gender;
  name: FullName;
  email: string;
};

type NewBaseUser = BaseUser & {
  password: string;
};

export type NewTalent = NewBaseUser;

export type NewEmployer = NewBaseUser & {
  company: string;
};

export type SignupFormValues = (NewTalent | NewEmployer) & {
  type: UserType;
};
```
that should fix the issue